### PR TITLE
Scaling and UV flip fixed.

### PR DESCRIPTION
**Pull request by ludvigarlebrink**

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Editor/GUI/Editors/ModelEditor.cpp [hunk 1]
```diff
@@ -46,16 +46,29 @@ void ModelEditor::Show() {
...
         if (hasSourceFile) {
             ImGui::Text("Mesh Data");
+            ImGui::Checkbox("Uniform Scaling", &uniformScaling);
+
+            if (uniformScaling) {
+                float uniScale = scale.x;
+                ImGui::DragFloat("Scale", &uniScale, 0.01f);
+                scale = glm::vec3(uniScale);
+            }
+            else
```
Comment (andens) [inline src/Editor/GUI/Editors/ModelEditor.cpp]: Please put the else keyword on the same line as the previous brace and wrap the else body in braces according to style guidelines.

File: src/Editor/Util/AssetConverter.cpp [hunk 2]
```diff
@@ -103,33 +99,34 @@ void AssetConverter::ConvertMesh(aiMesh * aMesh, Geometry::AssetFileHandler * fi
...
     // Positions.
     if (aMesh->HasNormals()) {
         for (unsigned int i = 0; i < numVertices; ++i) {
             Geometry::CpyVec(vertices[i].position, aMesh->mVertices[i]);
+            vertices[i].position *= scale;
         }
     } else {
         success = false;
         errorString.append("WARNING: The model has no positions yet the user is trying to import them.\n");
-        for (unsigned int i = 0; i < numVertices; ++i) {
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: We don't have any explicit mentioning about braces, but I would expect them to be present similarly to if statements in the style example.

File: src/Editor/Util/AssetConverter.cpp [hunk 3]
```diff
@@ -103,33 +99,34 @@ void AssetConverter::ConvertMesh(aiMesh * aMesh, Geometry::AssetFileHandler * fi
...
     if (aMesh->HasTextureCoords(0)) {
         for (unsigned int i = 0; i < numVertices; ++i) {
             Geometry::CpyVec(vertices[i].textureCoordinate, aMesh->mTextureCoords[0][i]);
+            if (flipUVs) 
+                vertices[i].textureCoordinate.y = 1.0f - vertices[i].textureCoordinate.y;
         }
     } else {
         success = false;
         errorString.append("WARNING: The model has no texture coordinates yet the user is trying to import them.\n");
-        for (unsigned int i = 0; i < numVertices; ++i) {
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: Braces, perhaps.

File: src/Editor/Util/AssetConverter.cpp [hunk 4]
```diff
@@ -140,81 +137,74 @@ Video::Geometry::VertexType::StaticVertex * AssetConverter::ConvertStaticVertice
     } else {
         success = false;
         errorString.append("WARNING: The model has no normals yet the user is trying to import them.\n");
-        for (unsigned int i = 0; i < numVertices; ++i) {
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: Braces, perhaps.

File: src/Editor/Util/AssetConverter.cpp [hunk 5]
```diff
@@ -140,81 +137,74 @@ Video::Geometry::VertexType::StaticVertex * AssetConverter::ConvertStaticVertice
...
+        for (unsigned int i = 0; i < numVertices; ++i)
             vertices[i].normal = glm::vec3(1.0f, 0.0f, 0.0f);
-        }
     }
 
     // Tangents.
     if (aMesh->HasTangentsAndBitangents()) {
-        for (unsigned int i = 0; i < numVertices; ++i) {
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: Braces, perhaps.

File: src/Editor/Util/AssetConverter.cpp [hunk 6]
```diff
@@ -140,81 +137,74 @@ Video::Geometry::VertexType::StaticVertex * AssetConverter::ConvertStaticVertice
...
+        for (unsigned int i = 0; i < numVertices; ++i)
             Geometry::CpyVec(vertices[i].tangent, aMesh->mTangents[i]);
-        }
     } else {
         success = false;
         errorString.append("WARNING: The model has no tangents yet the user is trying to import them.\n");
-        for (unsigned int i = 0; i < numVertices; ++i) {
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: Braces, perhaps.

File: src/Editor/Util/AssetConverter.cpp [hunk 7]
```diff
@@ -140,81 +137,74 @@ Video::Geometry::VertexType::StaticVertex * AssetConverter::ConvertStaticVertice
...
     // Positions.
     if (aMesh->HasNormals()) {
         for (unsigned int i = 0; i < numVertices; ++i) {
             Geometry::CpyVec(vertices[i].position, aMesh->mVertices[i]);
+            vertices[i].position *= scale;
         }
     } else {
         success = false;
         errorString.append("WARNING: The model has no positions yet the user is trying to import them.\n");
-        for (unsigned int i = 0; i < numVertices; ++i) {
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: Braces, perhaps.

File: src/Editor/Util/AssetConverter.cpp [hunk 8]
```diff
@@ -140,81 +137,74 @@ Video::Geometry::VertexType::StaticVertex * AssetConverter::ConvertStaticVertice
...
+        for (unsigned int i = 0; i < numVertices; ++i)
             vertices[i].position = glm::vec3(1.0f, 0.0f, 0.0f);
-        }
     }
 
     // Texture coordinates.
     if (aMesh->HasTextureCoords(0)) {
         for (unsigned int i = 0; i < numVertices; ++i) {
             Geometry::CpyVec(vertices[i].textureCoordinate, aMesh->mTextureCoords[0][i]);
+            if (flipUVs)
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: Braces around body according to style example.

File: src/Editor/Util/AssetConverter.cpp [hunk 9]
```diff
@@ -140,81 +137,74 @@ Video::Geometry::VertexType::StaticVertex * AssetConverter::ConvertStaticVertice
...
+                vertices[i].textureCoordinate.y = 1.0f - vertices[i].textureCoordinate.y;
         }
     } else {
         success = false;
         errorString.append("WARNING: The model has no texture coordinates yet the user is trying to import them.\n");
-        for (unsigned int i = 0; i < numVertices; ++i) {
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: Braces, perhaps.

File: src/Editor/Util/AssetConverter.cpp [hunk 10]
```diff
@@ -140,81 +137,74 @@ Video::Geometry::VertexType::StaticVertex * AssetConverter::ConvertStaticVertice
...
+        for (unsigned int i = 0; i < numVertices; ++i)
             vertices[i].textureCoordinate = glm::vec2(0.0f, 0.0f);
-        }
     }
 
     // Normals.
     if (aMesh->HasNormals()) {
-        for (unsigned int i = 0; i < numVertices; ++i) {
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: Braces, perhaps.

File: src/Editor/Util/AssetConverter.cpp [hunk 11]
```diff
@@ -140,81 +137,74 @@ Video::Geometry::VertexType::StaticVertex * AssetConverter::ConvertStaticVertice
...
+        for (unsigned int i = 0; i < numVertices; ++i)
             Geometry::CpyVec(vertices[i].normal, aMesh->mNormals[i]);
-        }
     } else {
         success = false;
         errorString.append("WARNING: The model has no normals yet the user is trying to import them.\n");
-        for (unsigned int i = 0; i < numVertices; ++i) {
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: Braces, perhaps.

File: src/Editor/Util/AssetConverter.cpp [hunk 12]
```diff
@@ -140,81 +137,74 @@ Video::Geometry::VertexType::StaticVertex * AssetConverter::ConvertStaticVertice
...
+        for (unsigned int i = 0; i < numVertices; ++i)
             vertices[i].normal = glm::vec3(1.0f, 0.0f, 0.0f);
-        }
     }
 
     // Tangents.
     if (aMesh->HasTangentsAndBitangents()) {
-        for (unsigned int i = 0; i < numVertices; ++i) {
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: Braces, perhaps.

File: src/Editor/Util/AssetConverter.cpp [hunk 13]
```diff
@@ -140,81 +137,74 @@ Video::Geometry::VertexType::StaticVertex * AssetConverter::ConvertStaticVertice
...
+        for (unsigned int i = 0; i < numVertices; ++i)
             Geometry::CpyVec(vertices[i].tangent, aMesh->mTangents[i]);
-        }
     } else {
         success = false;
         errorString.append("WARNING: The model has no tangents yet the user is trying to import them.\n");
-        for (unsigned int i = 0; i < numVertices; ++i) {
```
Comment (andens) [inline src/Editor/Util/AssetConverter.cpp]: Braces, perhaps.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L177](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ac2484f0a56275315f5c7d0c2a80fb4ac6e58b58/src/Engine/Hymn.cpp#L177): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by ludvigarlebrink** [comment]
After a dice roll with chainsawkitten, the code standard is without bracers.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L175](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d9ded30da9c33ba65c9893866f44cc71a6d5b84d/src/Engine/Hymn.cpp#L175): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L177](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d52661ab1f83ad17c9a8cfe3c7edfb391376a01b/src/Engine/Hymn.cpp#L177): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**